tests/utils.py

```diff
@@ -183,3 +186,17 @@ def extract_fn_name(fn):
     if not match:
         raise RuntimeError
     return match.group("name")
+
+
+def suppress_deprecation_warning(*categories: Type[Warning]):
+    if not categories:
+        categories = (UserWarning,)
+    def deprecation_decor(test_fn):
+        @functools.wraps(test_fn)
+        def wrapper(*args, **kwargs):
+            for category in categories:
+                with warnings.catch_warnings():
+                    warnings.filterwarnings("ignore", category=category, module=r'.*pystiche')
+                    return test_fn(*args, **kwargs)
+        return wrapper
+    return deprecation_decor
```

Comment:
1. Since we named the function `suppress_deprecation_warning`, IMO it is fair game to not make it configurable to suppress all kind of warnings. Thus, I would remove the outer wrapper.
2. Since you are invoking `test_fn` inside the `categories` loop, you will effectively run the test multiple times and only suppress one warning category each run. We want to register multiple filters based on the `categories`, but only run the test once.
3. The `module` regex is a little to permissive IMO. It allows all modules that have the name `pystiche` in it. I suggest we use `module="pystiche([.].*)?"`. With that `module` can either be `pystiche` or any of its submodules `pystiche.*`

```suggestion
def suppress_deprecation_warning(test_fn):
    @functools.wraps(test_fn)
    def wrapper(*args, **kwargs):
        with warnings.catch_warnings():
            for category in (UserWarning, DeprecationWarning):
                warnings.filterwarnings(
                    "ignore", category=category, module="pystiche([.].*)?"
                )

            return test_fn(*args, **kwargs)

    return wrapper
```